signalbot/bot.py

```diff
@@ -220,17 +221,45 @@ def _resolve_receiver(self, receiver: str) -> str:
             "This should never happen."
         )
 
+    # see https://stackoverflow.com/questions/55184226/catching-exceptions-in-individual-tasks-and-restarting-them
+    @classmethod
+    async def _rerun_on_exception(cls, coro, *args, **kwargs):
+        """Restart coroutine by waiting an exponential time deplay"""
+        max_sleep = 5 * 60  # sleep for at most 5 mins until rerun
+        reset = 3 * 60  # reset after 3 minutes running successfully
+        init_sleep = 1  # always start with sleeping for 1 second
+
+        next_sleep = init_sleep
+        while True:
+            start_t = int(time.time())  # seconds
```

Comment:
I'd use `time.monotonic()` instead of `time.time()` to be safe. It's recommended for timers as it's unaffected by time zone or clock changes.

Reply:
great comment, thanks